projects/plugins/jetpack/modules/search/instant-search/store/actions.js

```diff
@@ -18,14 +18,16 @@ export function makeSearchRequest( options ) {
  * @param {object} params - Input parameters.
  * @param {object} params.options - Action options that generated this API response.
  * @param {object} params.response - API response.
+ * @param {object} params._id - Sequential ID used to determine recency of the response.
  *
  * @returns {object} Action object.
  */
-export function recordSuccessfulSearchRequest( { options, response } ) {
+export function recordSuccessfulSearchRequest( { options, response, _id } ) {
```

Comment:
Is there an existing convention to use `_id`? If not, I reckon `responseId` would be more descriptive.

Reply:
I wonder if the sequence should be added when sending out requests rather than already when they are responded? Because when the response is back, they've already have their natural order no matter we add the sequence or not.

Reply:
I wonder if the sequence should be added when sending out requests rather than already when they are responded? Because when the response is back, they've already have their natural order no matter we add the sequence or not.

Reply:
> Because when the response is back, they've already have their natural order no matter we add the sequence or not.

Good catch, this approach is going to require an overhaul.

Reply:
Fixed!

Reply:
Excellent catch @kangzj 👍